### PR TITLE
fix: close stderr in [run_in_directory]

### PR DESCRIPTION
**Pull request by rgrinberg**
Otherwise we're going to leak a file descriptor

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

File: ocaml-lsp-server/src/ocaml_lsp_server.ml [hunk 1]
```diff
@@ -929,6 +929,7 @@ let run_in_directory ~prog ~prog_is_quoted:_ ~args ~cwd ?stdin ?stdout ?stderr (
   in
   Unix.close stdin;
   if should_close_stdout then Unix.close stdout;
+  Option.iter stderr ~f:Unix.close;
```
Comment (voodoos) [inline ocaml-lsp-server/src/ocaml_lsp_server.ml:932]: Are you trying to fix an issue you observed ? Looking at Merlin's codebase I don't think there is any usage of `run_in_directory` where `stderr` is not `None`.

But yes we should close that file descriptor when stderr it redirected to a file.

**Review by voodoos — APPROVED** [review]
Looks good to me. Should I re-released with the fix ? I don't think it was ever happening since `stderr` is always `None`.

File: ocaml-lsp-server/src/ocaml_lsp_server.ml (hunk 1 above)
Comment (rgrinberg) [reply ocaml-lsp-server/src/ocaml_lsp_server.ml:932]: Yes, I never observed this issue but I imagine it could be possible in the future.

**Comment by rgrinberg** [comment]
No need to re-release, but it's a nice fix to have in case future versions of merlin use stderr for something

**Pull request merged**